src/EditorFeatures/VisualBasic/LineCommit/CommitBufferManager.vb

```diff
@@ -156,10 +141,35 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.LineCommit
         End Sub
 
         Private Structure FormattingInfo
+            ' This is the original flag to enable semantic formating on a span
             Public UseSemantics As Boolean
+            ' This is new flag that allows semantic formating of a single statement
+            Public UseSemanticStatementFormatting As Boolean
+            ' This is the total span to format, the span may or may not be semantically formated
             Public SpanToFormat As SnapshotSpan
+            ' This is a span that will always be semantically formated
+            Public SpanToSemanticFormat As SnapshotSpan
```

Comment:
can this not just be nullable?  if it's set, then we should be semantically formatted.  do we need UseSemanticStatementFormatting?

Reply:
Yes